app/lib/widgets/settings/share_settings.dart

```diff
@@ -0,0 +1,287 @@
+                                success!,
+                                style: const TextStyle(color: Colors.green),
+                              )
+                            : Container(),
+                      ],
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ShareHolderList extends StatefulWidget {
+  const _ShareHolderList({
+    required this.shareHolders,
+    required this.refreshList,
+    super.key,
+  });
+
+  final dynamic shareHolders;
+  final Function() refreshList;
+
+  @override
+  State<_ShareHolderList> createState() => _ShareHolderListState();
+}
+
+class _ShareHolderListState extends State<_ShareHolderList> {
+  get api => obtainApiClient()!;
+  String? error;
+  String? success;
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.shareHolders == null) {
+      return Center(
+          child: Column(children: const [
+        CircularProgressIndicator(),
+        Text("Loading shares"),
+      ]));
+    }
+
+    if (widget.shareHolders.length == 0) {
+      return const Text('No share holders');
+    }
+
+    return Container(
+      child: Column(
+        mainAxisSize: MainAxisSize.max,
+        children: [
+          error != null
+              ? Text(
+                  error!,
+                  style: TextStyle(color: Theme.of(context).errorColor),
+                )
+              : Container(),
+          success != null
+              ? Text(
+                  success!,
+                  style: const TextStyle(color: Colors.green),
+                )
+              : Container(),
+          Expanded(
+            child: ListView.builder(
+              itemCount: widget.shareHolders.length,
+              itemBuilder: (context, index) {
+                var shareHolder = widget.shareHolders[index];
+                var duration = shareHolder["duration"];
+                return ListTile(
+                  title: Text(shareHolder["email"]),
+                  subtitle: Text(duration == null
+                      ? "Forever"
+                      : (duration ~/ 29030400) > 1
+                          ? "${duration ~/ 29030400} years"
+                          : (duration ~/ 2419200) > 1
+                              ? "${duration ~/ 2419200} months"
+                              : (duration ~/ 604800) > 1
+                                  ? "${duration ~/ 604800} weeks"
+                                  : (duration ~/ 86400) > 1
+                                      ? "${duration ~/ 86400} days"
+                                      : (duration ~/ 3600) > 1
+                                          ? "${duration ~/ 3600} hours"
+                                          : "${duration ~/ 60} minutes"),
+                  trailing: IconButton(
+                    icon: const Icon(Icons.delete),
+                    onPressed: () async {
```

Comment:
Can you change this to a method on the class?
I don't think we should have code so far nested.